### PR TITLE
ServiceOffering ordering

### PR DESCRIPTION
**Pull request by slemrmartin**
Provides ordering of offerings with or without plan(form)

- old way
  - `POST /service_plans/:service_plan_id/order "{ \"service_parameters\": {...}", \"provider_controler_parameters\":{}}"`
- new way
  - `POST /service_offerings/:service_offering_id/order "{\"provider_controler_parameters\":{}}`
  - `POST /service_offerings/:service_offering_id/order "{\"service_parameters\": {...}", \"provider_controler_parameters\":{}}"`

**depends on**:
- [x] https://github.com/ManageIQ/topological_inventory-openshift/pull/99
- [x] https://github.com/ManageIQ/topological_inventory-ansible_tower/pull/24

**dependents**:
https://github.com/ManageIQ/topological_inventory-api-client-ruby/pull/16

### CODE REVIEW

**Comment by slemrmartin** [comment]
@Ladas please :eyes:

**Comment by slemrmartin** [comment]
cc @Hyperkid123

File: app/controllers/api/v1/service_offerings_controller.rb [hunk 1]
```diff
@@ -3,6 +3,48 @@ module V1
     class ServiceOfferingsController < ApplicationController
       include Api::V1::Mixins::IndexMixin
       include Api::V1::Mixins::ShowMixin
+      include Api::V1::Mixins::SourcesApiMixin
+
+      def order
+        service_offering = model.find(params_for_order[:service_offering_id].to_i)
```
Comment (Ladas) [inline app/controllers/api/v1/service_offerings_controller.rb]: @agrare do we always include `tenant_id => XY` into a query?

Comment (agrare) [reply app/controllers/api/v1/service_offerings_controller.rb]: I believe acts_as_tenant automatically adds the current tenant to the query but we should confirm

Comment (slemrmartin) [reply app/controllers/api/v1/service_offerings_controller.rb]: probably yes, I don't have tenancy disabled on localhost

Comment (agrare) [reply app/controllers/api/v1/service_offerings_controller.rb]: Should be easy to confirm by checking the rails log for the query it executes

Comment (slemrmartin) [reply app/controllers/api/v1/service_offerings_controller.rb]:
```
SELECT  "service_offerings".* FROM "service_offerings" WHERE "servic
e_offerings"."tenant_id" = $1 AND "service_offerings"."id" = $2 LIMIT $3  [["tenant_id", 1], ["id", 3], ["LIMIT", 1]]
```

Comment (Ladas) [reply app/controllers/api/v1/service_offerings_controller.rb]: awesome, I wasn't sure if find was covered

Comment (agrare) [reply app/controllers/api/v1/service_offerings_controller.rb]: :heart: perfect, lgtm

**Pull request merged**